pkg/autodiscovery/providers/kube_endpoints.go

```diff
@@ -95,6 +103,8 @@ func (k *KubeEndpointsConfigProvider) Collect() ([]integration.Config, error) {
 			continue
 		}
 		generatedConfigs = append(generatedConfigs, generateConfigs(config.tpl, kep)...)
+		endpointsID := apiserver.EntityForEndpoints(config.namespace, config.name, "")
+		k.monitoredEndpoints[endpointsID] = true
```

Comment:
do we need to check if the `endpointsID`  is already in the map before setting it to `true` ?

Reply:
I don't think so, the maps is used instead of a slice (for fast lookup) and we never set a value to `false`.
Even if `endpointsID` is already in the map its value can't be other than `true`. (which cannot happen because we delete the entry from the map when a service is added or deleted).